src/super_gradients/training/kd_trainer/kd_trainer.py

```diff
@@ -211,6 +211,7 @@ def _load_checkpoint_to_model(self):
         Initializes teacher weights with teacher_checkpoint_path if needed, then handles checkpoint loading for
          the entire KD network following the same logic as in Trainer.
         """
+        super(KDTrainer, self)._load_checkpoint_to_model()
```

Comment:
Is there a specific reason you move the call up ?

Reply:
The base `_load_checkpoint_to_model` loads checkpoints, but also sets some checkpoints attributes which are used by `KDTrainer._load_checkpoint_to_model`.